setup.cfg

```
@@ -45,7 +45,7 @@ zip_safe = False
 packages = sphinx_rtd_theme
 python_requires = >=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,!=3.5.*
 install_requires = 
-	sphinx >=1.6,<7
+	sphinx >=1.6,<8
```

I'd be 👍 on just dropping the upper bound, and letting users install the theme on newer Sphinx, and report display issues as they come up. It would stop this from blocking people's upgrades in the future. 
```suggestion
	sphinx >=1.6
```

---

I'm still hesitant to do anything like this. With docutils, Sphinx, and jinja releases, we've learned that having upper bounds would solve many dependency issues. Unfortunately, if there is an issue between our theme and Sphinx, it most likely will not result in an error or failure, but silently broken output.

These tend to turn into fires we have to put out too, as we can't easily/consistently retroactively apply these boundaries to dependencies for releases.

---

True, there's no obviously correct option, unfortunately :(

---

FWIW, as a consumer:

We pin all of our dependencies, so we can easily roll-back if a Sphinx bump ends up conflicting with the theme. When the theme has the pin, we don't really have a choice, we have to wait, even if it would work fine.

That said, I think the best case would be if the theme had the cap but was able to update shortly after Sphinx releases. I realize that's kind of an unbounded commitment though :-)

---

Seconding this. As the consumer, it is my responsibility to pin the deps I'm using. Strongly prefer when libraries only have a minimum restriction on their own deps, and no max (particularly in the absence of the ability to update the max quickly and release).

> We pin all of our dependencies, so we can easily roll-back if a Sphinx bump ends up conflicting with the theme. When the theme has the pin, we don't really have a choice, we have to wait, even if it would work fine.

---

I think @agjohnson -> it has to be done sooner or later, there many, many, many more problems with dependencies limited upper-bound for your consumers than not limting them. Basically by limiting Sphinx now you simply don't know of the problems and your users can't even help you with solving them by pointing out to issues. 

Suggestion: Release an RC  (or beta, whatever) version with Sphinx 7 removed and some basic testing done and ask all people commenting here to test it and report problems to you and even provide fixes. I would be personally super happy if there is one place where we coudl discuss those based on an RC.

I think maintainers of rtd_theme can observe enough of the willpower and community ready to make the move that if you simply ask for help on solving the problems, you will get it.

This is a great thing BTW. that you are releasing it for free and keeping it running - this is great and big thanks for years of doing it. And you have huge community that will help, if only they have a chance to.

---

> We pin all of our dependencies
> Seconding this. As the consumer, it is my responsibility to pin the deps I'm using.

This is great! We really wish more projects did do this and it should be the default really. But unfortunately for us, many projects still do not pin dependencies -- both transitive and direct dependencies even. Some of this is general packaging awareness, but it also mostly stems from years of Read the Docs magically deciding dependencies for projects. Doing so definitely has led to projects not defining important dependencies, and relying on our build process to do this instead.

So while this is a annoyance for projects that do perform package maintenance well, strict pinning is meant to avoid build failures for all of the projects that don't define direct and transitive dependencies explicitly.

> I think @agjohnson -> it has to be done sooner or later

Yeah, I definitely agree with all of this too, I want to see this package limit it's dependencies a great deal. But there is indeed no correct answer for all users.

> Suggestion: Release an RC (or beta, whatever) version with Sphinx 7 removed and some basic testing done

Yup, absolutely. I'm currently on other projects, so it's a bit more up to other core team to cut releases right now, but we have been trying to aim for frequent, small stakes releases and more RC releases. Especially if we are avoiding taking on features on these releases, they are very low stakes, but still require an annoying amount of testing.

---

>it also mostly stems from years of Read the Docs magically deciding dependencies for projects. Doing so definitely has led to projects not defining important dependencies, and relying on our build process to do this instead.

I wonder if it would be reasonable for Read the Docs to manage the version pin in _its_ build process?

---

> I wonder if it would be reasonable for Read the Docs to manage the version pin in its build process?

Hah well this is exactly what we originally thought when we decided on this sort of magical approach to dependencies. But we're actually trying to move away from this pattern and stop holding user's hands through packaging configuration. Teach a person to fish and all that.

It's definitely a slow process though, and not exactly pain free, but does help projects (and surely our scope of concerns) in the long run.